charts/opensearch/templates/statefulset.yaml

```diff
@@ -319,8 +319,14 @@ spec:
         ports:
         - name: http
           containerPort: {{ .Values.httpPort }}
+          {{- if .Values.hostHttpPort }}
+          hostPort: {{ .Values.hostHttpPort }}
```

Comment:
I understand the requirement but may I know why are you using this approach for exposing OpenSearch? From what I know the host IP can change when the container is restarted, two containers using the same hostPort cannot be scheduled on the same node and the usage of the hostPort is considered a privileged operation on OpenShift.

Reply:
Ok I try to explain: We need only exposure of transportHostPort. I added also httpHostPort for analogy.
Why do we need to expose tranportHostPort?
We have two datacenter and need to use Cross-Cluster Replication Plugin to replicate index data. Each datacenter has 5 nodes reserved for opensearch replicas. (5 replicas for master/data per Opensearch-Cluster we deploy with antiAffinity="hard"). So for seed ip address there are only the 5 ips of the nodes in each datacenter with the exposed transportHostPort. There is some psp which allows us to use hostPort in K8s-Cluster.
To setup proper opensearch-cluster-node communication on transport level we set also transport.publish_host and transport.publish_port in extraEnvs.

Reply:
Use case makes sense to me